bidict/_bidict.py

```diff
@@ -7,58 +7,75 @@ class bidict(BidirectionalMapping, MutableMapping):
     """
     Mutable bidirectional map type.
     """
+
     def _del(self, key):
         val = self._fwd[key]
         del self._fwd[key]
         del self._bwd[val]
 
     def __delitem__(self, key):
+        """
+        Analogous to dict.__delitem__(), keeping bidirectionality intact.
+        """
         self._del(key)
 
     def __setitem__(self, key, val):
-        self._put(key, val)
+        """
+        Inserts the given mapping iff it wouldn't overwrite an existing key
```

Comment:
@lordmauve Tweaked wording in dcba3aba4c564f92b32e035f69228e2a96dbfd25, let me know if that's less confusing.

Added pep257 compliance in 8bf96068af3d8befce274bc80d1f2047bcb31480. Thanks for flagging.